retriever/lib/datasets.py

```diff
@@ -1,14 +1,21 @@
-from retriever.lib.scripts import SCRIPT_LIST, get_script
-
+from retriever.lib.scripts import SCRIPT_LIST, get_script, get_dataset_names_upstream
+from retriever.lib.defaults import RETRIEVER_REPOSITORY
 
 def datasets(keywords=None, licenses=None):
     """Search all datasets by keywords and licenses."""
     script_list = SCRIPT_LIST()
-
     if not keywords and not licenses:
-        return sorted(script_list, key=lambda s: s.name.lower())
+        # The scripts present locally
+        offline_scripts = sorted(script_list, key=lambda s: s.name.lower())
+        # The scripts present in upstream retriever repository
+        online_retriever_script_names = get_dataset_names_upstream(repo=RETRIEVER_REPOSITORY)
```

Comment:
The variable names are still lone. Use smaller names for these group of variables. 
The comments should help in telling us what the variable name holds

eg. upstream_scripts, upstream_online, native_scripts, local_scripts, recipes_scripts,